File: dialog-cli/Sources/DialogCLI/Models/QuestionModels.swift
import Foundation

struct QuestionOption: Codable {
    let label: String
    let description: String?
}

enum QuestionType: String, Codable {
    case choice
    case text
}

struct QuestionItem: Codable {
    let id: String
    let question: String
    let type: QuestionType
    let options: [QuestionOption]
    let multiSelect: Bool
    let placeholder: String?

    init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        id = try container.decode(String.self, forKey: .id)
        question = try container.decode(String.self, forKey: .question)
        type = try container.decodeIfPresent(QuestionType.self, forKey: .type) ?? .choice
        options = try container.decodeIfPresent([QuestionOption].self, forKey: .options) ?? []
        multiSelect = try container.decodeIfPresent(Bool.self, forKey: .multiSelect) ?? false
        placeholder = try container.decodeIfPresent(String.self, forKey: .placeholder)
    }
}

// Answer can be either choice indices or text
enum QuestionAnswer {
    case choices(Set<Int>)
    case text(String)

    var isEmpty: Bool {
        switch self {
        case .choices(let set): return set.isEmpty
        case .text(let str): return str.isEmpty
        }
    }
}

struct QuestionsRequest: Codable {
    let questions: [QuestionItem]
    let mode: String  // "wizard" | "accordion"
    let position: DialogPosition
}

struct QuestionsResponse: Codable {
    let dialogType: String
    let answers: [String: StringOrStrings]
    let cancelled: Bool
    let dismissed: Bool
    let completedCount: Int
    let snoozed: Bool?
    let snoozeMinutes: Int?
    let remainingSeconds: Int?
    let feedbackText: String?
    let instruction: String?
}

File: dialog-cli-windows/Models/QuestionModels.cs
namespace DialogCLI.Models;

public enum QuestionType
{
    Choice,
    Text,
}

public record QuestionOption(
    string Label,
    string? Description
);

public record QuestionItem(
    string Id,
    string Question,
    QuestionType Type = QuestionType.Choice,
    QuestionOption[]? Options = null,
    bool MultiSelect = false,
    string? Placeholder = null
)
{
    // Ensure Options is never null for choice-type questions
    public QuestionOption[] Options { get; init; } = Options ?? [];
};

public record QuestionsRequest(
    QuestionItem[] Questions,
    string Mode,
    DialogPosition Position
);

public class QuestionsResponse
{
    public string DialogType { get; set; } = "questions";
    public Dictionary<string, StringOrStrings> Answers { get; set; } = new();
    public bool Cancelled { get; set; }
    public bool Dismissed { get; set; }
    public int CompletedCount { get; set; }
    public bool? Snoozed { get; set; }
    public int? SnoozeMinutes { get; set; }
    public int? RemainingSeconds { get; set; }
    public string? FeedbackText { get; set; }
    public string? Instruction { get; set; }
}
